<a href="https://colab.research.google.com/github/Manikanta-204/Play-store-app-review-analysis/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

In [3]:
# Import numpy and pandas packages for the data analysis purpose
import numpy as np
import pandas as pd
# Import matplotlib and seaborn packages for the data visualization purpose
import matplotlib.pyplot as plt             # pyplot is a module
import seaborn as sns
#%matplotlib inline                         # %matplotlib inline sets the backend of matplotlib to the 'inline' backend


In [4]:
# Mounting google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [103]:
# Define the path where data is kept
file_path = '/content/drive/MyDrive/Colab Notebooks/Play store app review analysis EDA Capstone project/Play Store Data.csv'

In [104]:
# Reading the csv format dataset
playstore_app_df = pd.read_csv(file_path)

In [105]:
# Checking the number of rows and columns in the playstore app dataset
playstore_app_df.shape

(10841, 13)

There are total 10841 rows and 13 columns are present in the playstore app 
dataset

In [106]:
# print the first 5 records with all columns in dataset
playstore_app_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [107]:
# printing the last 5 records from playstore app file
playstore_app_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [108]:
# Checking the what are the coulumns names that are present in the dataset
playstore_app_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [109]:
# Get the information about our playstore app dataset
playstore_app_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


It displays the total number of rows and columns, Datatypes of each column and memory requirement

In [110]:
# By default describe method displays the statistics of numerical columns 
playstore_app_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [111]:
# Checking the  null/missing values if those are present in our dataset
playstore_app_df.isnull().sum()


App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [112]:
playstore_app_df.shape

(10841, 13)

In [113]:
# Removing the null/missing values 

# playstore_app_df.dropna(how = 'any',axis=0,inplace = True)    
# If we use above code we can remove all null values but we can remove only maximum null value column present in our dataset 

playstore_app_df.dropna(how = 'any' ,subset= ['Rating'],axis=0,inplace = True)

# Again verify for any missing values are present in Rating column
playstore_app_df['Rating'].isnull().sum()

0

In [114]:
# We can replace minimum amount of null values with most common repeated values that are present in the same column 
playstore_app_df['Current Ver'].fillna(playstore_app_df['Current Ver'].mode()[0],inplace = True)
playstore_app_df['Android Ver'].fillna(playstore_app_df['Android Ver'].mode()[0],inplace = True)

In [115]:
playstore_app_df['Current Ver'].isnull().sum()
playstore_app_df['Android Ver'].isnull().sum()

0

In [116]:
playstore_app_df.loc[:,['Current Ver','Android Ver']]

Current Ver         Android Ver
0                   1.0.0        4.0.3 and up
1                   2.0.0        4.0.3 and up
2                   1.2.4        4.0.3 and up
3      Varies with device          4.2 and up
4                     1.1          4.4 and up
...                   ...                 ...
10834               1.0.0          4.1 and up
10836                1.48          4.1 and up
10837                 1.0          4.1 and up
10839  Varies with device  Varies with device
10840  Varies with device  Varies with device

[9367 rows x 2 columns]

In [117]:
#
playstore_app_df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    1
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64

In [118]:
playstore_app_df.shape

(9367, 13)